starknet_py/net/account/account.py

```diff
@@ -0,0 +1,401 @@
+            max_fee=0,
+            version=self._version,
+            nonce=nonce,
+            contract_address=self.address,
+        )
+
+        max_fee = await self._get_max_fee(transaction, max_fee, auto_estimate)
+
+        return dataclasses.replace(transaction, max_fee=max_fee)
+
+    async def _verify_message_hash(self, msg_hash: int, signature: List[int]) -> bool:
+        """
+        Verify a signature of a given hash
+
+        :param msg_hash: hash to be verified
+        :param signature: signature of the hash
+        :return: true if the signature is valid, false otherwise
+        """
+        calldata = [msg_hash, len(signature), *signature]
+
+        call = Call(
+            to_addr=self.address,
+            selector=get_selector_from_name("is_valid_signature"),
+            calldata=calldata,
+        )
+        try:
+            await self._client.call_contract(call=call, block_hash="pending")
+            return True
+        except ClientError as ex:
+            if re.search(r"Signature\s.+,\sis\sinvalid", ex.message):
+                return False
+            raise ex
+
+    async def _signed_estimate_fee(
+        self,
+        tx: Union[InvokeFunction, Declare, DeployAccount],
+        block_hash: Optional[Union[Hash, Tag]] = None,
+        block_number: Optional[Union[int, Tag]] = None,
+    ) -> EstimatedFee:
+        """
+        :param tx: Transaction which fee we want to calculate
+        :return: Estimated fee
+        """
+        signature = self.signer.sign_transaction(tx)
+        tx = _add_signature_to_transaction(tx, signature)
+
+        return await self._client.estimate_fee(
+            tx=tx,
+            block_hash=block_hash,
+            block_number=block_number,
+        )
+
+    async def get_nonce(self) -> int:
+        """
+        Get the current nonce of the account
+
+        :return: nonce
+        """
+        return await self._client.get_contract_nonce(
+            self.address, block_number="pending"
+        )
+
+    async def get_balance(
+        self, token_address: Optional[AddressRepresentation] = None
+    ) -> int:
+        """
+        Checks account's balance of specified token.
+
+        :param token_address: Address of the ERC20 contract.
+            If not specified it will be payment token (wrapped ETH) address.
+        :return: Token balance
+        """
+
+        token_address = token_address or self._get_default_token_address()
+
+        low, high = await self._client.call_contract(
+            Call(
+                to_addr=parse_address(token_address),
+                selector=get_selector_from_name("balanceOf"),
+                calldata=[self.address],
+            ),
+            block_hash="pending",
+        )
+
+        return (high << 128) + low
+
+    async def sign_invoke_transaction(
+        self,
+        calls: Calls,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+        **kwargs,
```

Comment:
This is a workaround I'm not 100% happy about: I want to remove `version` parameter from these methods that was present in `AccountClient` because it doesn't make sense to have it if the account has a version anyway. 

So I've added `**kwargs` so that `version` can still be passed and handled for the time being (only used in `_AccountClient`.

Reply:
But perhaps the more future proof approach would be to not have `**kwargs` here and just ignore typechecker in few places it's complaining?

Reply:
Totally agree. We should design it for the new `Account` not the old `AccountClient`

Reply:
Let's do that then,